source/Nevermore/Advanced/ReadTransaction.cs

```diff
@@ -538,11 +539,13 @@ public async Task<DbDataReader> ExecuteReaderAsync(PreparedCommand preparedComma
         PreparedCommand PrepareLoadMany<TDocument, TKey>(IEnumerable<TKey> idList)
         {
             var mapping = configuration.DocumentMaps.Resolve(typeof(TDocument));
-            var tableName = mapping.TableName;
+
+            if (mapping.IdColumn.Type != typeof(TKey))
```

Comment:
Thanks for adding this check, should we include a test for it?

Reply:
Sure, I will add all tests for this kind of check in other places where we use `PrepareXXXXXX` command.